src/Nethermind/Nethermind.Serialization.Rlp/WithdrawalDecoder.cs

```diff
@@ -43,7 +43,7 @@ public class WithdrawalDecoder : IRlpStreamDecoder<Withdrawal>, IRlpValueDecoder
             Index = decoderContext.DecodeULong(),
             ValidatorIndex = decoderContext.DecodeULong(),
             Address = decoderContext.DecodeAddress(),
-            Amount = decoderContext.DecodeUInt256()
+            AmountInGWei = decoderContext.DecodeULong()
```

Comment:
Different methods when reading the same `ulong` value: `DecodeUlong` vs `DecodeULong`. This should be addressed to avoid any confusion in the future.